From e6d1e12c705294f251b88c426191c3391943756f Mon Sep 17 00:00:00 2001
From: Steve Cornelius <steve.cornelius@freescale.com>
Date: Fri, 23 Mar 2012 09:32:01 -0700
Subject: [PATCH 1384/2463] ENGR00180943-3: Add CAAM instantiation to i.MX6Q
 ARM2 board platform

Add CAAM instantiation to i.MX6Q ARM2 board platform configuration.
No other device-on-board configurations added at this time.

Signed-off-by: Steve Cornelius <steve.cornelius@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c |    2 ++
 arch/arm/mach-mx6/devices-imx6q.h   |    4 ++++
 2 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 1c00882..967212a 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -2082,6 +2082,8 @@ static void __init mx6_arm2_init(void)
 
 	imx6q_add_imx_snvs_rtc();
 
+	imx6q_add_imx_caam();
+
 	imx6q_add_imx_i2c(0, &mx6_arm2_i2c0_data);
 	imx6q_add_imx_i2c(1, &mx6_arm2_i2c1_data);
 	i2c_register_board_info(0, mxc_i2c0_board_info,
diff --git a/arch/arm/mach-mx6/devices-imx6q.h b/arch/arm/mach-mx6/devices-imx6q.h
index 2c3e0c5..995a5bd 100644
--- a/arch/arm/mach-mx6/devices-imx6q.h
+++ b/arch/arm/mach-mx6/devices-imx6q.h
@@ -27,6 +27,10 @@ extern const struct imx_snvs_rtc_data imx6q_imx_snvs_rtc_data __initconst;
 #define imx6q_add_imx_snvs_rtc()	\
 	imx_add_snvs_rtc(&imx6q_imx_snvs_rtc_data)
 
+extern const struct imx_caam_data imx6q_imx_caam_data __initconst;
+#define imx6q_add_imx_caam()	\
+	imx_add_caam(&imx6q_imx_caam_data)
+
 extern const struct imx_anatop_thermal_imx_data
 imx6q_anatop_thermal_imx_data __initconst;
 #define imx6q_add_anatop_thermal_imx(id, pdata)	\
-- 
1.7.7.4

